File: 250430/1차원 윷놀이/yutnori-1d.cpp
#include <iostream>
#include <vector>
using namespace std;

int n, m, k;
int nums[12];
int result = 0;

void func(int curr, vector<int>& v) {
    if (curr > n) {
        int cnt = 0;
        for (int val : v) {
            if (val >= m) cnt++;
        }
        result = max(result, cnt);
        return;
    }

    for (int i = 1; i <= k; i++) {
        v[i] += nums[curr];
        func(curr + 1, v);
        v[i] -= nums[curr];
    }
}

int main() {
    cin >> n >> m >> k;
    vector<int> v(k + 1, 1);

    for (int i = 1; i <= n; i++) {
        cin >> nums[i];
    }
    func(1, v);
    cout << result;

    return 0;
}

File: 250807/1차원 윷놀이/yutnori-1d.py
n, m, k = map(int, input().split())
nums = list(map(int, input().split()))
horses = [1] * k
ans = 0

# k개의 말에 대하여 각각이 n개의 경우에 할당된다. O(k*n)
# 목표지점을 넘어가는 말은 이동 중단
def calc():
    scores = 0
    for i in range(k):
        if horses[i] >= m:
            scores += 1
    return scores

def backtracking(curr):
    global ans
    if curr == n:
        ans = max(ans, calc())
        return
    else:
        for i in range(k):
            if horses[i] >= m:
                continue
            horses[i] += nums[curr]
            backtracking(curr + 1)
            horses[i] -= nums[curr]

backtracking(0)
print(ans)